tensorboard/plugins/graph/tf_graph_board/tf-graph-board.ts

```diff
@@ -293,7 +294,15 @@ class TfGraphBoard extends LegacyElementMixin(PolymerElement) {
   @property({type: Object})
   handleEdgeSelected: object;
   fit() {
-    (this.$.graph as any).fit();
+    ((this.$.graph as unknown) as TfGraph).fit();
+  }
+  async downloadAsImage(filename: string) {
+    const blob = await ((this.$.graph as unknown) as TfGraph).getImageBlob();
+    const element = document.createElement('a');
+    element.href = URL.createObjectURL(blob);
+    element.download = filename;
+    element.click();
```

Comment:
I’ve tested this and it seems to work for me, but I’m a bit surprised.
I expected a synthetic `e.click()` event to not trigger a download.
Do you know what I’m misunderstanding?

Reply:
My understanding is that `el.click()`, while synthetic, still triggers the browser's "default behavior", as well as any 'click' listeners registered on the element.  I'm not sure that really answers the root of your question, I looked briefly at the spec but didn't see anything definitive.  Similar to `download`, calling `.click` on other types of elements may trigger native behaviors such as page navigation, opening the browser file selector dialog, submitting a form, etc.